api/src/convert/block_metadata.rs

```diff
@@ -24,13 +29,27 @@ impl TryFrom<&blockchain::BlockMetadataContents> for BlockMetadataContents {
     fn try_from(src: &blockchain::BlockMetadataContents) -> Result<Self, Self::Error> {
         let block_id = src.get_block_id().try_into()?;
         let quorum_set = src.get_quorum_set().try_into()?;
-        let report = src.get_verification_report().try_into()?;
+        let attestation_evidence = match (src.has_dcap_evidence(), src.has_verification_report()) {
+            (true, false) => {
+                let evidence = src.get_dcap_evidence().try_into()?;
+                AttestationEvidence::DcapEvidence(evidence)
+            }
+            (false, true) => {
+                let report = src.get_verification_report().try_into()?;
+                AttestationEvidence::VerificationReport(report)
+            }
+            _ => {
+                return Err(ConversionError::MissingField(
+                    "attestation_evidence".to_string(),
+                ))
+            }
```

Comment:
While technically this captures `(true, true)` that shouldn't happen with a one of